run/session-handling/src/main/java/com/example/gettingstarted/util/FirestoreSessionFilter.java

```diff
@@ -0,0 +1,179 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.gettingstarted.util;
+
+import com.example.gettingstarted.actions.HelloWorldServlet;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.common.collect.Maps;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.SecureRandom;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Logger;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+@WebFilter(
+    filterName = "FirestoreSessionFilter ",
+    urlPatterns = {""})
+public class FirestoreSessionFilter implements Filter {
+  private static final SimpleDateFormat dtf = new SimpleDateFormat("yyyyMMddHHmmssSSS");
+  private static final Logger logger = Logger.getLogger(HelloWorldServlet.class.getName());
+  private static Firestore firestore;
+  private static CollectionReference sessions;
+
+  // [START sessions_handling_init]
+  @Override
+  public void init(FilterConfig config) {
+    // Initialize local copy of datastore session variables.
+    firestore = FirestoreOptions.getDefaultInstance().getService();
+    sessions = firestore.collection("sessions");
+
+    try {
+      // Delete all sessions unmodified for over two days.
+      Calendar cal = Calendar.getInstance();
+      cal.setTime(new Date());
+      cal.add(Calendar.MINUTE, -2);
+      cal.getTime();
+      Date twoDaysAgo = cal.getTime();
```

Comment:
`getInstance()` already defaults to now, and MINUTE -> DAY
```suggestion
      Date twoDaysAgo = Calendar.getInstance().add(Calendar.DAY, -2).getTime();
```

Reply:
Changing this slightly because the `.add()` method modifies the date and returns void :|. Also `Calendar` doesn't have a `DAY` constant, using 48 hours instead.